### ISSUE TITLE
Middle name field to be added to signed consent and HIPAA forms in PWA

### ISSUE THREAD

**Issue by cunnaneaq** [episphere/connectApp#537]
The middle name field is missing from the signature coordinates in the signed consent and HIPAA forms in the PWA

The middle name field is optional but if a participant chooses to enter it when they sign it should display exactly how they signed it with their middle name. Here are the CIDs, want to also make sure they are the name fields extracted from consent (vs the name fields from User Profile)</p>

471168198 | First Name extracted from consent form |   | First Name | RcrtCS_Fname_v1r0
736251808 | Last Name extracted from consent form |   | Last Name | RcrtCS_Lname_v1r0
436680969 | Middle Name extracted from consent form |   | Middle Name | RcrtCS_Mname_v1r0

**Comment by jhflorey** [comment]
@cunnaneaq I noticed the consent form and HIPAA form still had the middle name field. Please help re-check again.
![image](https://github.com/episphere/connectApp/assets/24559107/1d4b4e44-1a7b-4a7b-b471-336183b2ef70)
![image](https://github.com/episphere/connectApp/assets/24559107/8e7314d1-9352-4d65-90db-9bc8129fae10)
![image](https://github.com/episphere/connectApp/assets/24559107/0aed9632-f0d3-42a6-b588-9f62321f4b00)

**Comment by jhflorey** [comment]
@Davinkjohnson there doesn't seem to be no any consent form with lower caase "v".

**Comment by cunnaneaq** [comment]
@jhflorey we want to keep the middle name field on the consent screen, but we'd like for the middle name to render on the signed PDF version of the consent and HIPAA forms if a participant enters a middle name.  Happy to check it out to see what the current situation is

**Comment by cunnaneaq** [comment]
@jhflorey thanks for thinking of this!  The middle name should also render on the signed data destruction and HIPAA revocation forms if the participant types in a middle name.

**Comment by cunnaneaq** [comment]
Consent and HIPAA Authorization forms - first, middle, and last names rendering correctly on signed PDFs

HIPAA Revocation and Data Destruction forms - first, middle, and last names seem to be pulling from consent signature.  Please use these concept IDs for the  HIPAA revocation and data destruction forms.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/cunnaneaq/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/cunnaneaq/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">



765336427 | First Name   extracted from HIPAA revocation form
-- | --
479278368 | Last Name   extracted from HIPAA revocation form
826240317 | Middle Name   extracted from HIPAA revocation form



</body>

</html>

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/cunnaneaq/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/cunnaneaq/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">



104278817 | First Name   extracted from Data Destruction form
-- | --
744604255 | Last Name   extracted from Data Destruction form
268665918 | Middle Name   extracted from Data Destruction form



</body>

</html>

**Comment by brotzmanmj** [comment]
They should also display the Suffix from the respective signatures for all four forms.

**Comment by jhflorey** [comment]
Got it @cunnaneaq  @brotzmanmj

**Comment by jhflorey** [comment]
@brotzmanmj Will the Suffix be displayed at the end of signature? Example `FirstName MiddleName LastName Suffix`

**Comment by brotzmanmj** [comment]
Yes, that's correct

**Comment by jhflorey** [comment]
@cunnaneaq @brotzmanmj Please check it in dev

**Comment by cunnaneaq** [comment]
@jhflorey consent, HIPAA authorization, data destruction, and HIPAA revocation forms all look great!  Middle names and suffixes are appearing correctly.  Ready for stage

**Comment by brotzmanmj** [comment]
@jhflorey  Kaitlyn checked this on the SMDB and we don't see these updates there. Can you check if they have been applied there or only on the PWA?

**Comment by jhflorey** [comment]
@brotzmanmj Got it.

**Comment by robertsamm** [comment]
This looks good in dev for middle name appearing correctly in the SMDB on all forms. @kmazzilli will also be confirming this when she completes her next round of testing in dev for data destruction.

**Comment by robertsamm** [comment]
Hello, sorry I forgot to test the suffix in the above test. I retested and am still seeing issues with the suffix in the SMDB. See the 3 screenshots below.

- Consent/HIPAA forms missing suffix in ‘signature’ 
- Revocation form missing suffix in ‘signature’
- Destruction form missing suffix in ‘signature’ and suffix undefined in ‘printed name’


Consent/HIPAA forms on SMDB:
 
![image](https://github.com/episphere/connectApp/assets/67074735/cdc6b96e-3e2d-4b3d-8b4f-b23973445531)

Revocation on SMDB:
 
![image](https://github.com/episphere/connectApp/assets/67074735/4f7c0415-ff26-4355-a4c1-f8fe70627bbb)

Destruction on SMDB:
![image](https://github.com/episphere/connectApp/assets/67074735/b10baf7d-75d8-4b8a-8255-f588d51bb348)

**Comment by jhflorey** [comment]
Could you please provide test participant connect_id

**Comment by robertsamm** [comment]
Test pt for suffix in dev Connect ID: 2215090924

**Comment by robertsamm** [comment]
Hi @jhflorey is this ready for retest?

**Comment by Davinkjohnson** [comment]
Hi @robertsamm sorry for the delay, Jessica is OOO today. She did put in a fix last night for the Suffix. You should be able to test again.

**Comment by robertsamm** [comment]
Hello, the signature fields still aren't properly being pulled in on the revocation and destruction forms in the SMDB. It looks like middle name and suffix is correct for both forms, but first name and last name look to be pulling from the consent not the separate revocation and destruction name variables. Please be sure to check all the signature fields on the destruction and revocation forms again before sending it back to us for testing.

See screenshots below from test pt Connect ID: 5490689165 

Incorrect on SMDB Rev:
![image](https://github.com/episphere/connectApp/assets/67074735/c42fd428-2591-441b-b351-ff1ee64f4e4c)
Incorrect on SMDB Destr: 
![image](https://github.com/episphere/connectApp/assets/67074735/7539225b-27c7-4748-9862-35bdcd96eb13)


Correct on PWA:
PWA Revocation:
![image](https://github.com/episphere/connectApp/assets/67074735/6b11249a-7e41-428a-bcf0-7a9736da6bae)
PWA Destr:
![image](https://github.com/episphere/connectApp/assets/67074735/1d8f808e-3e29-4048-a18a-289193876b7a)

**Comment by jhflorey** [comment]
@robertsamm as the requirement of this issue , it only refers to adding a middle name. I will update the first and last name for now but I think we need to write the requirement for completeness next time.

**Comment by brotzmanmj** [comment]
Hi @jhflorey sorry for the confusion. I added the requirement above: "They should also display the Suffix from the respective signatures for all four forms." three weeks ago. Let me know in the future if there is a better way to delineate updated scope within an issue and will be happy to structure it as such.

**Comment by jhflorey** [comment]
@robertsamm  yup, but it's for PWA. IMHO we should create the same issue for https://github.com/episphere/dashboard in the next time.

**Comment by jhflorey** [comment]
@brotzmanmj @robertsamm just merged my code changes into dev.

**Comment by robertsamm** [comment]
@jhflorey Just tested this again and it looks good everywhere. Names+suffixes look good on all forms (consent, HIPAA, revocation, destruction) in both PWA and SMDB. @kmazzilli will test one more time when she does the data destruction testing today but this looks good to go to stage on my end. Thanks Jessica!

**Comment by robertsamm** [comment]
Tested in stage and looks good. Ready for prod

**Comment by jhflorey** [comment]
Hi @robertsamm. Please help check this issue in stage. I've already checked my test participant in stage(3800067442). That worked as my expectation.

**Comment by robertsamm** [comment]
Hi @jhflorey I tested this in stage already. It looked good to me as well, ready for prod.
